Python/ast.c

```diff
@@ -1666,6 +1703,34 @@ ast_for_decorated(struct compiling *c, const node *n)
     return thing;
 }
 
+static expr_ty
+ast_for_namedexpr(struct compiling *c, const node *n)
+{
+    /* if_stmt: 'if' namedexpr_test ':' suite ('elif' namedexpr_test ':' suite)*
+         ['else' ':' suite]
+       namedexpr_test: test [':=' test]
+       argument: ( test [comp_for] |
+            test ':=' test |
+            test '=' test |
+            '**' test |
+            '*' test )
+    */
+    expr_ty target, value;
+
+    target = ast_for_expr(c, CHILD(n, 0));
```

Comment:
Seems `(x) := 1` is accepted. Is it desirable? If it is not, see `ast_for_call` which makes `f(x=1)` allowed, but rejects `f((x)=1)`.